action.yaml

```diff
@@ -0,0 +1,117 @@
+name: Install MDAnalysis
+description: Install MDAnalysis
+inputs:
+  version:
+    description: >
+      Version of MDAnalysis to install.
+      Can be a specified version (e.g. 2.0.0),
+      "develop", or "latest".
+    required: true
+    default: latest
+  install-tests:
+    description: Whether to install MDAnalysisTests
+    required: true
+    default: false
+  installer:
+    description: >
+      Installer to use.
+      Must be one of pip, conda, or mamba.
+    required: true
+    default: pip
+  shell:
+    description: >
+      Which shell to use.
+      If you have a conda environment, you should
+      use `bash -l {0}`.
+      If you are using pip, you should use `bash`,
+      as `bash -l {0}` can fail on Mac runners.
+    required: true
+    default: bash
+outputs:
+  installed-version:
+    description: Installed MDAnalysis version
+    value: "${{ steps.print-version.outputs.installed-version }}"
+
+
+runs:
+  using: "composite"
+  steps:
+    - name: Uninstall pre-existing versions
+      shell: ${{ inputs.shell }}
+      run: |
+        # uninstall tests even if not reinstalling,
```

Comment:
I know that successive pip / conda uninstall/install calls are kinda broken (something about registry or another). Might be worth explaining that the action can be flaky if used multiple times in a workflow?

Reply:
I could keep uninstalling until none? pip uninstall is fairly foolproof I thought, although might have to do multiple times

Reply:
iirc the issue is a bit deeper than that, https://github.com/pypa/pip/issues/9129 (there's some other issue iirc but I can't find it).

In any case, I don't think this is wrong, it just might be worth just adding in the readme that combining multiple rounds of conda & pip uninstalls & installs could lead to undefined behaviour is some cases?

Reply:
Ah. Yes, that's why I use pip to uninstall things 🙃

I can add it to the README, but tbh this action is not my top priority at the moment. It's "public" but I'd rather not advertise it loudly and focus on more pressing issues in other repos for now. I'm assuming people in MDAKits won't be willy-nilly un- and re-installing MDA multiple times in their CI with different tools, or at least they really shouldn't be.

Reply:
That's not an assumption I'd make, if we can think of a way to make it fail, someone _will_ get it done.